walk-in-interview/src/main/java/com/google/job/data/JobsDatabase.java

```diff
@@ -59,6 +59,17 @@
                 .set(job);
     }
 
+    /**
+     * Marks a job post as DELETED.
+     *
+     * @param jobId Cloud Firestore Id of the job post.
+     * @return A future of the detailed information of the update.
+     */
+    public Future<WriteResult> markJobPostAsDeleted(String jobId) {
+        DocumentReference documentReference = FireStoreUtils.getFireStore().collection(JOB_COLLECTION).document(jobId);
+        return documentReference.update(JOB_STATUS_FIELD, JobStatus.DELETED);
```

Comment:
I actually think this should be done in a transaction. right now your database class is delegating the existence check for jobId to the caller. Typically you would do a transaction on document(jobId), if the document doesn't exist you should fail the transaction or something, if it exists you can update the status field.
https://firebase.google.com/docs/firestore/manage-data/transactions

This same comment is valid for anything mutating a job by Id. You start a transaction, verify that the job exists, then do the mutation (or fail).